Landing Page / Registration / Login Frontend

muhammedyusuf-sermet opened the pull request:
The changes in this branch include the initialization of our front end with React and helper libraries (material-ui) and the frontend design of the the page combining the registration and login and acts as the landing page of the application. The branch is currently stable.

Backend of the registration and login functionality is a work-in-progress by @malisit. Once the functionality is completed, the appropriate webhooks will be added to Login and Register buttons. The tests will be added when the functionality is added for registration/login.

@hekemp requesting your review since you were also looking at the front end

Related Issues: #29 #36

hekemp commented on frontend/src/renderer/components/Footer.tsx:
```diff
@@ -0,0 +1,11 @@
+export const Footer: React.StatelessComponent<{}> = () => {
+  return (
+    <div className="footer-container">
+      <h5>© D&D Tools 2019 - All rights reserved</h5>
```
Minor note, but our project is DM Tools rather than D&D Tools

muhammedyusuf-sermet replied:
Fixed, will commit with other changes.

hekemp commented on frontend/src/renderer/components/Registration.tsx:
```diff
@@ -0,0 +1,78 @@
+	        <Grid item xs={12}>
+	          <FormControlLabel
+	            control={<Checkbox color="secondary" name="addMailingList" value="yes" />}
+	            label="Keep me updated on the D&D Community!"
```
We don't have a mailing list so i'm not sure if this is quite necessary

muhammedyusuf-sermet replied:
I thought it as a way to get consent to send them e-mails. I'll get rid of it for now, we can always add it back if needed as it's not a major code change.

hekemp commented:
We probably also want to check that a user isn't logged in before showing them the log in page

muhammedyusuf-sermet commented:
> 
> 
> We probably also want to check that a user isn't logged in before showing them the log in page

That's partly in the scope of the login functionality Muhammed is working on, skipping this check for this commit.

> 
> 
> This looks nice, though I do have two concerns.
> 
>     1. I don't know if we can (legally) use the Dungeons and Dragons logo. It may help to use the "DM-Tools" logo that I added to docs (_I believe_) just to avoid legal complications? Hopefully that's not a huge change.
Changed that with our DM-Tools logo.
> 
>     2. @aszecsei has requested that everything in master be tested, and so that means that there should be code to test the front end even if there's no backend functionality to go with it.
As discussed in Discord, I'll add a snapshot test.

muhammedyusuf-sermet commented:
After long hours of debugging and researching, we now have a working testing environment for the frontend with the correct dependencies and configurations. Just a note for our future-selves: testing stateless components are fishy, don't go that route unless necessary... They are better to use for smaller static components that can be incorporated into larger components (and get tested inside these larger components with states).

All issues are addressed in this review. However, since there were some changes made to our CI in #119, I'll have to resolve the conflicts and make it work with the Travis updates. After that we can merge this one.

muhammedyusuf-sermet commented:
@hekemp The branch is good to go. Feel free to review the new changes and to finalize the merge. Thanks!

muhammedyusuf-sermet commented:
Screenshot:

![image](https://user-images.githubusercontent.com/14010967/52926479-9507b080-32fb-11e9-9bef-457cfebbe02f.png)
